custom_components/tuya_local/devices/smartmi_e1_purifier.yaml

```diff
@@ -0,0 +1,56 @@
+name: Smartmi Air Purifier E1
+products:
+  - id: pwrmggdhwkz1lnvh
+    name: Smartmi E1
+primary_entity:
+  entity: fan
+  translation_key: fan_with_presets
+  icon: "mdi:air-purifier"
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: true
+          icon: "mdi:air-purifier"
+        - dps_val: false
+          icon: "mdi:air-purifier-off"
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: sleep
+          value: Sleep
```

Comment:
values that are using translations should be all lower case to match the keys.  

Even though "sleep" will map to itself, you still need to list the mapping here, since this list is used to tell HA which preset modes are available.

Reply:
Also, icons listed here are for the entity, not for the specific option.  Since there is no icon_priority, it will be random whether the preset_mode icon or the switch icon above is applied.

Changing icons dynamically interferes with themes and user customisation, so should be used sparingly, only when there is useful information conveyed. (again, this is another thing that you will see overused in older configs)